common/src/main/scala/net/psforever/objects/serverobject/structures/Amenity.scala

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2017 PSForever
+package net.psforever.objects.serverobject.structures
+
+import net.psforever.objects.serverobject.PlanetSideServerObject
+import net.psforever.types.PlanetSideEmpire
+
+/**
+  * Amenities are elements of the game that belong to other elements of the game.<br>
+  * <br>
+  * Normal `PlanetSideServerObject` entities (server objects) tend to have properties that are completely internalized.
+  * An `Amenity` is a server object that maintains a fixed association with another server object.
+  * This association strips away at the internalization and redirects a reference to some properties somewhere else.
+  * An `Amenity` object belongs to its `Owner` object;
+  * the `Amenity` objects looks to its `Owner` object for some of its properties.
+  * @see `FactionAffinity`
+  */
+abstract class Amenity extends PlanetSideServerObject {
+  private var owner : PlanetSideServerObject = Building.NoBuilding
+
+  def Faction : PlanetSideEmpire.Value = Owner.Faction
+
+  /**
+    * Reference the object that is in direct association with (is superior to) this one.
+    * @return the object associated as this object's "owner"
+    */
+  def Owner : PlanetSideServerObject = owner
+
+  /**
+    * Set an object to have a direct association with (be superior to) this one.
+    * @see `Amenity.AmenityTarget`
+    * @param obj the object trying to become associated as this object's "owner"
+    * @tparam T a validation of the type of object that can be an owner
+    * @return the object associated as this object's "owner"
+    */
+  def Owner_=[T : Amenity.AmenityTarget](obj : T) : PlanetSideServerObject = {
```

Comment:
Why is this syntax used?

Why not use `def Owner_=(obj : Amenity.AmenityTarget) : PlanetSideServerObject = {`

Edit: I think I get it. You're using the generic type T, but you're saying that although T is generic, it still needs to be a subtype of Amenity.AmenityTarget.